content/posts/building_jnigen.md

```diff
@@ -0,0 +1,262 @@
+---
+title: "jnigen - Building JNI bindings generator for Dart Language"
+type: "post"
+date: 2023-06-29T07:49:21+05:30
+draft: true
+tags: [dart, flutter, jnigen]
+showTableOfContents: true
+---
+>> Caine: Then why bother?
+>>
+>> John: Maybe I am wrong.
+>>
+> John Wick 4, (2023)
+
+It's been a long overdue to write a post about my past year's Google Summer of Code (GSoC) project. `jnigen` is an experimental bindings generator which aims to provide Java interoperability for Dart. It works by generating wrappers which call JNI through Dart's FFI (Foreign Function Interface).
+
+I developed the initial versions of this package under the guidance of Daco Harkes and Liam Appelbe from Dart language team. Hossein Yousefi from Dart team is developing the project further, adding many features such as Generics and Kotlin language support.
+
+## Why?
+
+The current way of accessing platform APIs on Flutter is Method Channels. Channels neatly avoid the intricacies of native language interop, by using an RPC like mechanism instead. However, there are a few drawbacks - all calls to method channels have to be asynchronous, and sharing memory is not possible. Perhaps more important, currently the use of method channels requires writing quite a bit of boilerplate. [^pigeon]
```

Comment:
I'm not familiar with the specifics of RPCs, but they don't have to be message passing right? Maybe just say "by using message passing instead".

You could mention the coding boilerplate and runtime overhead of serializing and deserializing data.

Reply:
These are loosely defined words I guess. Usually RPC seems to mean message passing to call methods etc..